### PR TITLE
Update temp/humidity in room #1316

### LINKED ISSUE #1316: Box "temperature in room" & "humidity in room" are not refreshing data with websockets
Data is not updated live on those boxes

### PR DESCRIPTION
**Pull request by atrovato**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- ~~[ ] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)~~
- ~~[ ] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.~~
- ~~[ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)~~
- ~~[ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).~~
- ~~[ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).~~

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Update temperature and humdity according to websocket event.

1. Load devices in room
2. Check if event matches device in room
3. If yes, reload data from API

⚠️ I didn't test it in real life

Fixes #1316

### CODE REVIEW

File: front/src/components/boxs/room-humidity/RoomHumidity.jsx [hunk 1]
```diff
@@ -47,8 +49,13 @@ class RoomHumidityBoxComponent extends Component {
     this.props.getHumidityInRoom(this.props.box, this.props.x, this.props.y);
   };
 
+  updateRoomHumidity = payload => {
+    this.props.deviceFeatureWebsocketEvent(this.props.x, this.props.y, payload);
+  };
+
   componentDidMount() {
     this.refreshData();
+    this.props.session.dispatcher.addListener(WEBSOCKET_MESSAGE_TYPES.DEVICE.NEW_STATE, this.updateRoomHumidity);
```
Comment (Pierre-Gilles) [inline front/src/components/boxs/room-humidity/RoomHumidity.jsx:58]: session is not defined here it seems

Comment (Pierre-Gilles) [reply front/src/components/boxs/room-humidity/RoomHumidity.jsx:58]: Screenshot 2022-03-15 at 15 41 25

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
I got a little crash on the RoomHumidity component ?

File: front/src/components/boxs/room-humidity/RoomHumidity.jsx (hunk 1 above)
Comment (atrovato) [reply front/src/components/boxs/room-humidity/RoomHumidity.jsx:58]: The same on temp box.
Thanks.
Now fixed.

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
I can't get the feature to work, is it working on your side?

I'm sending an event in MQTT, and the dashboard value is not refreshed

**Comment by atrovato** [comment]
No, that was blind dev, sorry.
I take it back, and will try to emulate it.

**Comment by atrovato** [comment]
I update this one with Cypress tests.
I used "example" service to fake setValue method. Is it OK for you?

PS: this is based on https://github.com/GladysAssistant/Gladys/pull/1488

**Comment by atrovato** [comment]
I created cypress tests, with websocket emulation.
I also used the "example" service to store device feature value in database, what do you think about this? @Pierre-Gilles 

For each box, 3 cypress tests:
 1. check empty box
 2. check box well updated with matching feature
 3. check box is not updated with non matching feature


humidity

temperature

**Review by Pierre-Gilles — APPROVED** [review]
Tested, works fine, thanks for this PR ! :)

**Pull request merged**